src/main/java/com/google/gwt/site/webapp/client/GWTProjectEntryPoint.java

```diff
@@ -443,25 +486,44 @@ private boolean isHomePage(String path) {
   }
 
   private void styleHomepage() {
-    final int windowHeight = $(window).height();
-    int sectionHeight = $("#letsbegin").height();
+    final int windowHeight = window.innerHeight;
+    int sectionHeight = ((HTMLElement) document.querySelector("#letsbegin")).offsetHeight;
 
     if (windowHeight > sectionHeight) {
-      $(".home section").each(new Function() {
-        @Override
-        public void f() {
-          $(this)
-              .css("height", windowHeight + "px")
-              .css("padding", "0");
-          GQuery container = $(this).find(".container");
-          container.css("padding-top", (windowHeight - container.height()) / 2 + "px");
-        }
+      forEach(".home section", element -> {
+        element.style.setProperty("height", windowHeight + "px");
+        element.style.setProperty("padding", "0");
+        HTMLElement container = (HTMLElement) element.querySelector(".container");
+        container.style.setProperty("padding-top", (windowHeight - container.offsetHeight) / 2 + "px");
       });
     }
   }
 
+  private void forEach(String selector, ElementConsumer consumer) {
+    document.querySelectorAll(selector).forEach((element, index, parent) -> {
+      consumer.accept((HTMLElement) element);
+      return null;
+    });
+  }
+
+  private void forEachChild(Element parent, String selector, ElementConsumer consumer) {
+    parent.childNodes.forEach((element, index, p) -> {
+      if (element instanceof HTMLElement) {
+        if (((HTMLElement) element).matches(selector)) {
```

Comment:
Element has querySelectorAll, why not use that?

Reply:
to keep the semantic of direct children matching a selector I figured this is a straightforward solution. It's possible to get the same from `querySlectorAll` using ':scope>...' as I've just learned, but concatenating selectors is maybe tricky...